File: akk_example/src/main/scala/com/hdu/actors/SecondActor.scala
package com.hdu.actors

import akka.actor.{Actor, Props}

/**
  * DESCRIPTION:
  *
  * @author shushoufu
  * @Date 2018/4/9
  * @Time 下午3:56
  */
object SecondActor{
  def props(name:String):Props = Props(classOf[SecondActor],name);
}

class SecondActor(name:String) extends Actor{

  override def preStart(): Unit = {
    println(s"the second actor is preStart, path is ${self.path}")
  }


  override def postStop(): Unit = {
    println(s"the second actor is postStop, path is ${self.path}")
  }

  override def receive: Receive = {
    case _ => println(String.format("this name is %s,sender path is %s",name,sender().path))
  }
}

File: akk_example/src/main/java/com/hdu/chapter1_createAndFind/SecondActor.java
package com.hdu.chapter1_createAndFind;

import akka.actor.AbstractActor;
import akka.actor.Props;

/**
 * DESCRIPTION:
 *
 * @author shushoufu
 * @Date 2018/6/12
 * @Time 下午2:36
 */
public class SecondActor extends AbstractActor {

    public static Props props() {
        return Props.create(SecondActor.class);
    }

    @Override
    public void postStop() throws Exception {
        super.postStop();
        System.out.println("second actor is stopped...");
    }

    @Override
    public void preStart() throws Exception {
        super.preStart();
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .matchEquals("print", (r) -> {
                    Thread.sleep(1000);
                    System.out.println("this is second actor:path is " + getSelf().path().toString() + ",sender is " + getSender().path().toString());

                })
                .match(Integer.class, (i) -> {
                    Thread.sleep(1000);

                    System.out.println("this is second actor:path is " + getSelf().path().toString() + ",sender is " + getSender().path().toString() + ", number is " + i);
                })
                .build();
    }
}
